internal/provider/kubernetes/httproute.go

```diff
@@ -121,17 +121,61 @@ func (r *httpRouteReconciler) Reconcile(ctx context.Context, request reconcile.R
 
 	log.Info("reconciling httproute")
 
-	// Fetch the HTTPRoute from the cache.
-	httpRoute := &gwapiv1b1.HTTPRoute{}
-	err := r.client.Get(ctx, request.NamespacedName, httpRoute)
-	if errors.IsNotFound(err) {
-		log.V(2).Info("httproute not found, deleting it from the ResourceTable")
+	// Fetch all HTTPRoutes from the cache.
+	routeList := &gwapiv1b1.HTTPRouteList{}
+	if err := r.client.List(ctx, routeList); err != nil {
+		return reconcile.Result{}, fmt.Errorf("error listing httproutes")
+	}
+
+	found := false
+	for i := range routeList.Items {
+		// Store the httproute in the resource map.
+		route := routeList.Items[i]
+		routeKey := NamespacedName(route.DeepCopy())
```

Comment:
Don't think you need to do a DeepCopy here:
```suggestion
		routeKey := NamespacedName(&route)
```